File: secure_token.php
<?php

$secret = "mysecret";
$uri = "/my/uri";
$expire = "1470055000";

$token = str_replace(["=", '+', '/'], ['', '-', '_'], base64_encode(md5($secret . $uri . $expire, true)));

echo "http://www.example.org" . $uri . "?st=" . $token . "&e=" . $expire . "\n";

?>

File: secure_token.java
package azion_secure_token;

import java.security.MessageDigest;
import java.security.NoSuchAlgorithmException;

import sun.misc.BASE64Encoder;

public class secure_token {

	
	public static void main(String[] args) throws NoSuchAlgorithmException {
    	
		String secretKey = "mysecret";	
		String uri = "/my/uri";
		long expire = 1470055000;
		
		
		String aValidar = secretKey + uri + expire;
		
		MessageDigest md = MessageDigest.getInstance("MD5");
		md.update(aValidar.getBytes());
		
		String token = new BASE64Encoder().encode(md.digest()).replace("=","").replace("+","-").replace("/","_");
		
		System.out.printf("http://www.example.org%s?st=%s&e=%s",uri,token,expire);
	}
}
